lib/Graph/PlaceholderBindings.cpp

```diff
@@ -44,7 +44,8 @@ bool PlaceholderBindings::compare(const PlaceholderBindings *A,
   for (const auto &phTensorPair : phMapA) {
     auto *placeholder = phTensorPair.first;
     const auto *tensorA = phTensorPair.second;
-    const auto *tensorB = B->get(placeholder);
+    const auto *tensorB =
+        B->get(B->getPlaceholderByName(placeholder->getName()));
```

Comment:
We need to talk about this.

This function was meant to compare two `PlaceholderBindings` and return `true` if the *same* `Placeholders` in both point to `Tensors` that are identical in value. But now that the `Executor` creates and owns some `Placeholders`, it is not possible to correctly mock expected `PlaceholderBindings` instances in `ExecutorTest` since those `Placeholders` are internal to the `Executor`. I worked around this by modifying this method to return true if `Placeholders` *with the same name* in both `PlaceholderBindings` instances point to `Tensors` that are identical in value, but I don't feel good about this.

Reply:
I think since we guarantee names are unique this will work, but it's another drop in the bucket of "names as unique identifiers is awkward".